druid/src/widget/textbox.rs

```diff
@@ -325,7 +325,8 @@ impl<T: TextStorage + EditableText> Widget<T> for TextBox<T> {
         if !self.suppress_adjust_hscroll && !self.multiline {
             self.update_hscroll(ctx.size().width, env);
         }
-        if ctx.env_changed() && self.placeholder.needs_rebuild_after_update(ctx) {
+        if self.multiline || (ctx.env_changed() && self.placeholder.needs_rebuild_after_update(ctx))
```

Comment:
I think the issue here is that editor's `update` method is calling `request_paint` instead of `request_layout`; maybe try making that change instead (`src/text/editor.rs:L166`) and see if it fixes your problem?